nemo_skills/pipeline/generate.py

```diff
@@ -272,8 +293,14 @@ def generate(
     with run.Experiment(expname) as exp:
         extra_arguments_original = extra_arguments
         # TODO: reduce code duplication
+        donefiles = defaultdict(list)
         if random_seeds:
             for chunk_id in chunk_ids:
+                for seed in random_seeds:
+                    single_output_dir = f"{output_dir}{'/generation' if generation_type == GenerationType.generate else ''}"
```

Comment:
how about we unify this? Let's always use top-level without "generation" subfolder as it's more convenient and the fact that currently folder structure is different is confusing

Reply:
one thing I find valuable about the current approach is--if I want to judge or reward model, I can put them all their subfolders in the same folder, e.g.,

```bash
experiment-step-01
     generation
     judgement
     reward
```

whereas, with the proposal, I would have to do something like:
```bash
experiment-step-01
    generate
        generation
    judgement
        generation
    reward
        generation
```

So the alternate proposal is to name the output directory of generation based on the generation type. I'm fine either way, ultimately. wdyt?

Reply:
why? What I propose is to remove the "generation", not to add it. Then you can still do what you want with subfolders, right?

Reply:
Oh okay yeah that works. Initially interpreted the making them all the same as with current "/generation". Much more clear now and agree.

Reply:
fixed in [75fa210](https://github.com/NVIDIA/NeMo-Skills/pull/350/commits/75fa210c8c134a7f2b8b0be5048e0fc0b017ee24)